proxy/Main.cc

```diff
@@ -1972,6 +1974,31 @@ mgmt_storage_device_cmd_callback(void *data, char *arg, int len)
   return NULL;
 }
 
+static void *
+mgmt_lifecycle_msg_callback(void *, char *data, int len)
+{
+  APIHook *hook = lifecycle_hooks->get(TS_LIFECYCLE_MSG_HOOK);
+  TSPluginMsg msg;
+  MgmtInt op;
+  MgmtMarshallString tag;
+  MgmtMarshallData payload;
+  static const MgmtMarshallType fields[] = {MGMT_MARSHALL_INT, MGMT_MARSHALL_STRING, MGMT_MARSHALL_DATA};
+
+  if (mgmt_message_parse(data, len, fields, countof(fields), &op, &tag, &payload) == -1) {
+    Error("[mgmt_lifecyle_msg_callback] RPC message parsing error.");
```

Comment:
Please avoid function names in user-level error messages (it can just be emitted with ``show_location``). Suggest "Failed to parse plugin message".